OpenRA.Mods.Common/Effects/Bullet.cs

```diff
@@ -164,8 +165,11 @@ public void Tick(World world)
 			if (info.ContrailLength > 0)
 				contrail.Update(pos);
 
-			if (ticks++ >= length || (info.Blockable && world.ActorMap
-				.GetUnitsAt(world.Map.CellContaining(pos)).Any(a => a.HasTrait<IBlocksProjectiles>())))
+			if (ticks++ >= length
+				|| (info.Blockable && world.ActorMap
+					.GetUnitsAt(world.Map.CellContaining(pos))
+					.Any(a => a.TraitsImplementing<BlocksProjectiles>()
+						.Any(b => !b.IsTraitDisabled))))
```

Comment:
This is a bit of a monster to have inside a conditional check, so could you please pull this out into an `IsBlockedByActor()` method?